confuse.py

```diff
@@ -1262,16 +1269,20 @@ def __repr__(self):
     def convert(self, value, view):
         """Check that the value is a string and matches the pattern.
         """
-        if isinstance(value, BASESTRING):
-            if self.pattern and not self.regex.match(value):
-                self.fail(
-                    u"must match the pattern {0}".format(self.pattern),
-                    view
-                )
-            return value
-        else:
+        if not isinstance(value, BASESTRING):
             self.fail(u'must be a string', view, True)
 
+        if self.pattern and not self.regex.match(value):
+            self.fail(
+                u"must match the pattern {0}".format(self.pattern),
+                view
+            )
```

Comment:
Refactored to use guard clauses, so the happy path remains at the top indentation level.